src/standard/standard.ios.ts

```diff
@@ -250,6 +250,7 @@ function createPKShippingMethod(method: StripeShippingMethod): PKShippingMethod
 }
 
 function createAddress(address: STPAddress): StripeAddress {
+  if (!address) return undefined;
```

Comment:
I assume you've tested that `undefined` is acceptable to return instead of `{}`?

Since the return result gets passed to the StripePaymentListener, which is implemented by the app, it seems like a good idea to modify that API to make it clear the address may be undefined:
`StripePaymentListener.provideShippingMethods(address?: StripeAddress): StripeShippingMethods`

I guess you could argue that you'd never need to provide shipping methods if you don't support shipping, but this change seems innocuous and safer. You'd need to make the change in both `standard.common.ts` and `index.d.ts`.

I think you'd want to make a similar change to `StripeBackendAPI.completeCharge` since the address is passed there and this is implemented in the app. (What gets passed as the `shippingMethod` parameter in this case?)

Reply:
I've tested on iOS, can't test on Android per bug 2&3 in the https://github.com/triniwiz/nativescript-stripe/issues/63

Re: implementation - I was a bit torn because IMO `createAddress()` should not be called in the 1st place.  In lieu of a larger re-write, I decided to follow the same `undefined` pattern that is [already being used](https://github.com/triniwiz/nativescript-stripe/blob/master/src/standard/standard.ios.ts#L225).

IMO createAddress() should not be [inlined](https://github.com/triniwiz/nativescript-stripe/blob/master/src/standard/standard.ios.ts#L177).  Instead `paymentContextDidCreatePaymentResultCompletion()` should somehow check to see if `StripeConfig.shared().requiredShippingAddressFields` is set.  if not dont call it.

If you are cool with this approach, I can take a stab at writing the implementation. I'm hoping its just an easy `StripeConfig.shared().requiredShippingAddressFields` call inside `paymentContextDidCreatePaymentResultCompletion` to determine if add is required or not.